File: add/codesnippet/CSharp/c0b52280-ca91-42f3-a5ab-_1.cs
        public static ManualResetEvent allDone = 
            new ManualResetEvent(false);

        // handles the completion of the prior asynchronous 
        // connect call. the socket is passed via the objectState 
        // paramater of BeginConnect().
        public static void ConnectCallback1(IAsyncResult ar)
        {
            allDone.Set();
            Socket s = (Socket) ar.AsyncState;
            s.EndConnect(ar);
        }

File: add/codesnippet/CPP/c0b52280-ca91-42f3-a5ab-_1.cpp
   static ManualResetEvent^ allDone = gcnew ManualResetEvent( false );

   // handles the completion of the prior asynchronous 
   // connect call. the socket is passed via the objectState 
   // paramater of BeginConnect().
   static void ConnectCallback1( IAsyncResult^ ar )
   {
      allDone->Set();
      Socket^ s = dynamic_cast<Socket^>(ar->AsyncState);
      s->EndConnect( ar );
   }

